dev/source/docs/opendroneid.rst

```diff
@@ -0,0 +1,122 @@
+.. _common-opendroneid:
+
+===========
+OpenDroneID
+===========
+
+
+Building Firmware for Production
+================================
+
+ArduPilot provides methods for making firmware for autopilot systems integrating OpenDroneID Remote ID transmitter modules tamper-resistant as well as compliant with various countries initiatives.
+
+ArduPilot's tamper resistance currently consists of:
+
+- Using a special board ID for ODID (OpenDroneID) firmware that will reject normal attempts of loading firmware without the necessary functions and parameters required for compliance
+- Special READONLY parameters which can be set in the firmware during the compilation
+
+Future enhancements are planned.
+
+For an example see the hwdef and param files for an `ODID version of CubeOrange <https://github.com/ArduPilot/ardupilot/tree/master/libraries/AP_HAL_ChibiOS/hwdef/CubeOrange-ODID>`__ with a CAN OpenDroneID module integrated on the carrier board.
+
+.. note:: that the board ID has been changed different from normal firmware, to prevent loading it on a remoteid equipped system by a GCS. And the DID params have been set to read-only to prevent users from changing them and defeating the system.
+
+Building Firmware for Bench Testing and Experimentation
+=======================================================
+
+An experimental firmware version can be built for any autopilot by:
+
+- Adding the line "define AP_OPENDRONEID_ENABLED 1" to its hwdef.dat file, or simply building with the  waf build option, ``--enable-opendroneid``
+- Adding the waf build option, ``--enable-firmware-checking``, will add another tamper proofing check. If enabled, a bootloader will not run firmware with a different board ID. This extends the protection already afforded by using a unique
+
+
+Testing
+=======
+
+- Update MAVProxy/pymavlink to at least 1.8.54 (https://ardupilot.org/mavproxy/docs/getting_started/download_and_installation.html#updating)
+- In the directory you will be running MAVProxy from, create a remoteid.scr file with this content:
+
+.. code-block::
```

Comment:
This code block is not visible on the web page: 
https://ardupilot.org/dev/docs/opendroneid.html#testing

Reply:
fixed